Library/Homebrew/test/support/fixtures/failball.rb

```diff
@@ -0,0 +1,20 @@
+class Failball < Formula
+  def initialize(name = "failball", path = Pathname.new(__FILE__).expand_path, spec = :stable, alias_path: nil)
+    self.class.instance_eval do
+      stable.url "file://#{TEST_FIXTURE_DIR}/tarballs/testball-0.1.tbz"
+      stable.sha256 TESTBALL_SHA256
+    end
+    super
+  end
+
+  def install
+    prefix.install "bin"
+    prefix.install "libexec"
+
+    # This should get marshalled into a BuildError.
+    system "this_command_must_never_execute" if ENV["FAILBALL_BUILD_ERROR"]
```

Comment:
Can you explain this line a bit?

Reply:
Yup!

We're calling `system` with a bogus command to induce a failure, which causes it to raise a `BuildError` in the build process, per https://github.com/Homebrew/brew/blob/9bc17c8a82a36d9676528f761eaec3961f240cc8/Library/Homebrew/formula.rb#L1878

That `BuildError` contains a bunch of state that we use later on in analytics (and to tell the user what exactly went wrong), so we make a special case when catching that exception to re-add most of that state to the JSON blob that goes over the error pipe. The parent process reads that blob, realizes that it's a special case, and reconstructs a `BuildError` instead of a more generic `ChildProcessError`.

We do all of that only when `ENV["FAILBALL_BUILD_ERROR"]` is set in the test harness, so that we can fall through to the more generic test case below (when something other than a command fails in the build process, and a generic `ChildProcessError` gets reported back).

Reply:
Can this use `/usr/bin/false` instead?